### PR TITLE
[release-4.15] create a mechanism to deploy odf using operator-sdk

### PR DESCRIPTION
**Pull request by iamniting**
This is an manual cherry-pick of #394 #412

### CODE REVIEW

File: hack/install-odf.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+oc wait --timeout=5m --for jsonpath='{.status.phase}'=Succeeded -n "$INSTALL_NAMESPACE" csv $CSV_NAMES
+
+oc wait --timeout=5m --for condition=Available -n "$INSTALL_NAMESPACE" deployment \
+    csi-addons-controller-manager \
+    noobaa-operator \
+    ocs-operator \
+    odf-console \
+    odf-operator-controller-manager \
+    rook-ceph-operator
```
Comment (Nikhil-Ladha) [inline hack/install-odf.sh:31]: We have the `ux-backend-server` deployment in 4.15 as well.

File: .github/workflows/docker-build.yaml [hunk 2]
```diff
@@ -72,3 +72,5 @@ jobs:
       run: |
         docker run -d -p 5000:5000 --name registry registry:2
         make bundle-build bundle-push catalog-build
```
Comment (Nikhil-Ladha) [inline .github/workflows/docker-build.yaml:74]: We also need to update add the `catalog-deps-build` command here, right?

**Review by malayparida2000 — APPROVED** [review]
/lgtm

File: .github/workflows/docker-build.yaml (hunk 2 above)
Comment (Nikhil-Ladha) [reply .github/workflows/docker-build.yaml:74]: Also, to the `docker-push` action

Comment (iamniting) [reply .github/workflows/docker-build.yaml:74]: No there is no need for that, That is required only for the openshift CI

Comment (Nikhil-Ladha) [reply .github/workflows/docker-build.yaml:74]: Ok, I see we have that in 4.16 and main branch. So, shall we remove that from there as well?

Comment (iamniting) [reply .github/workflows/docker-build.yaml:74]: It's okay we can keep it there, It is not required at least on the old branch.

**Review by Nikhil-Ladha — APPROVED** [review]
/lgtm

File: .github/workflows/docker-build.yaml (hunk 2 above)
Comment (iamniting) [reply .github/workflows/docker-build.yaml]: I have backported that commit as well.

Comment (Nikhil-Ladha) [reply .github/workflows/docker-build.yaml]: Cool 😅

**Review by Nikhil-Ladha — COMMENTED** [review]
/lgtm

**Review by malayparida2000 — APPROVED** [review]

**Pull request merged**